docs/customization_extendibility.md

```diff
@@ -0,0 +1,72 @@
+
+## Conventions for defining additonal services
+When defining additional services for your project, it is recommended to follow these conventions to ensure your service is handled by `ddev` the same way the default services are.
+- Containers should be follow the naming convention `local-[sitename]-[servicename]`
+- Containers should be provided the following labels:
+  - `com.ddev.site-name: ${DDEV_SITENAME}`
+  - `com.ddev.approot: $DDEV_APPROOT`
+  - `com.ddev.app-url: $DDEV_URL`
+  - `com.ddev.container-type: [servicename]`
+- Exposing ports for service: you can expose the port for a service to be accessible as `sitename.ddev.local:portNum` while your project is running. This is achieved by the following for the container(s) being added:
+  - Define only the internal port in the `ports` section for docker-compose. The `externalPort:internalPort` convention normally used to expose ports in docker should not be used here, since we are leveraging the ddev router to expose the ports.
+  - Define the following environment variables in the `environment` section for docker-compose:
+    - `VIRTUAL_HOST=$DDEV_HOSTNAME`
+    - `VIRTUAL_PORT=portNum` To expose multiple ports for a single container, define the ports as comma-separated values.
+
+## Interacting with additional services
+Certain `ddev` commands, namely `ddev exec`, `ddev ssh`, and `ddev logs` interact with containers on an individual basis. By default, these commands interact with the web container for a project. All of these commands, however, provide a `--service` or `-s` flag allowing you to specify the service name of the container you want to interact with. For example, if you added a service to provide Apache Solr, and the service was named `solr`, you would be able to run `ddev logs --service solr` to retrieve the logs of the solr container.
+
+### Defining an additonal service - Apache Solr example
+The following is an example of an additional service that could be added to a project. This is a working example at time of writing. Annotations are added to highlight configurations important for running the service as part of a project in `ddev`. For full documentation on usage, see the [Docker Compose documentation](https://docs.docker.com/compose/overview/).
```

Comment:
No need to say "Is a working example at time of writing". Rather, can we make a test for it (or maybe we have one already?) And give a link to the file used in the test? Or maybe the test can use an examples/ directory here.